src/config.rs

```diff
@@ -9,10 +9,19 @@ use ldap_poller::{config::TLSConfig, AttributeConfig, CacheMethod, ConnectionCon
 use serde::Deserialize;
 use url::Url;
 
+/// App prefix for env var configuration
+const ENV_VAR_CONFIG_PREFIX: &str = "FAMEDLY_LDAP_SYNC";
+
 impl Config {
-	/// Read the config from a file
-	pub async fn from_file(path: &Path) -> Result<Self> {
-		let config: Config = serde_yaml::from_slice(&tokio::fs::read(path).await?)?;
+	/// Create new config from file and env var
+	pub fn new(path: &Path) -> Result<Self> {
```

Comment:
Let's not rename this - `::new` functions are usually expected to be quite simple (usually just assigning args to struct fields), this invokes IO and can fail.